packages/block-editor/src/hooks/gap.js

```diff
@@ -54,8 +55,12 @@ export function getGapBoxControlValueFromStyle( blockGapValue ) {
 
 	const isValueString = typeof blockGapValue === 'string';
 	return {
-		top: isValueString ? blockGapValue : blockGapValue?.top,
-		left: isValueString ? blockGapValue : blockGapValue?.left,
+		top: isValueString
+			? getSpacingPresetCssVar( blockGapValue )
+			: getSpacingPresetCssVar( blockGapValue?.top ),
+		left: isValueString
+			? getSpacingPresetCssVar( blockGapValue )
+			: getSpacingPresetCssVar( blockGapValue?.left ),
```

Comment:
This is not a blocker, and I'm not sure there's much to be done about it in this PR anyway.

When updating a single axial value we overwrite both top and left values:

```css
.wp-block-social-links.is-layout-flex {
    gap: var(--wp--preset--spacing--40) var(--wp--preset--spacing--20);
}
```

The fallback value differs between the editor and the frontend:

```css
.editor-styles-wrapper .wp-block-social-links.wp-container-37 {
    gap: 0 7px;
}
```

```
.wp-block-social-links.wp-container-10 {
    gap: 0.5em 7px;
}
```

Reply:
Oh, good catch. Let's look at adding some consistency there in a follow-up.

Reply:
Just thinking about this a little more, unfortunately, because we're concatenating to the single `gap` property we do need to include _some_ kind of fallback value there rather than letting the undefined side be undefined an inherit whatever's happening globally. Do you have an opinion on which one we should consolidate to? I'm a little torn 😅

Reply:
Ahhh I see. Maybe `0.5em` since it's already in layout.php and will show up on the frontend regardless?

I don't know either!

Reply:
Thanks @ramonjd, I was leaning slightly toward `0.5em` anyway. In this case, I think the fallback should only be used in the output of the Flex layout, to match the behaviour in the server-rendering (I think!), so in https://github.com/WordPress/gutenberg/pull/43296/commits/8bfbd94eeb1d58efc9666a3ab0bb70d76cb0aa8c I've updated the flex layout to _call_ the code with passing in `0.5em` — that way the fallback here is defined in the caller, so we can have other values for other potential future layout types.

Does that sound reasonable to you as a short-term fix?

Reply:
Sounds great, thanks for doing that!

Reply:
Works a treat, thanks!